src/lando/main/scm/git.py

```diff
@@ -278,8 +294,13 @@ def update_repo(self, pull_path: str, target_cset: Optional[str] = None) -> str:
             "+refs/heads/*:refs/remotes/origin/*",
             cwd=self.path,
         )
+
+        # Create a new work branch, named after the current time, to work in.
+        # Ideally, we'd use the revision number, too, but it's not available to the SCM.
+        # A date is good enough for now, if we need to dig into issues.
+        work_branch = f"lando-{datetime.now().strftime(ISO8601_TIMESTAMP_BASIC)}"
         self._git_run(
-            "checkout", "--force", "-B", branch, f"origin/{branch}", cwd=self.path
+            "checkout", "--force", "-B", work_branch, target_cset, cwd=self.path
         )
```

Comment:
Should we clean up after this? Otherwise we will accumulate a bunch of branches that will never be used again.

Reply:
Weight of history. It's what the standalone `HgRepo` object was using, and it got into the `abstract_scm`.

For git I'd use something like `base_commit`, but the meaning isn't changed.